### ISSUE TITLE
Holographic Remoting + MSBuildForUnity does not work if project name has spaces 

### ISSUE THREAD

**Issue by julenka** [microsoft/MixedRealityToolkit-Unity#6810]
## Describe the bug

Projects that have spaces in their name such as "New Unity Project" will cause errors when used with the MSBuildForUnity package, which is required to do Holographic Remoting

## To reproduce

1. Create a new project named "New Unity Project" and add mrtk 2.2 prerelease
2. Add MSBuildForUnity via the MixedRealityToolkit configuration window
3. Close and re-open Unity
4. Notice that editor will ber stuck in a compilation loop, with the following error text:

```
H:\New Unity Project\Assets\MixedRealityToolkit.Providers\WindowsMixedReality\DotNetAdapter> dotnet msbuild DotNetAdapter.csproj -restore  -v:minimal -p:NuGetInteractive=true  -t:Clean -p:Configuration=Release
UnityEngine.DebugLogHandler:Internal_Log(LogType, String, Object)
UnityEngine.DebugLogHandler:LogFormat(LogType, Object, String, Object[])
UnityEngine.Logger:Log(LogType, Object)
UnityEngine.Debug:Log(Object)
Microsoft.Build.Unity.<ExecuteMSBuildAsync>d__17:MoveNext() (at Library\PackageCache\com.microsoft.msbuildforunity@0.8.2\Editor\ProjectBuilder\MSBuildProjectBuilder.cs:403)
System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1:Start(<ExecuteMSBuildAsync>d__17&)
Microsoft.Build.Unity.MSBuildProjectBuilder:ExecuteMSBuildAsync(String, String, String, IProgress`1, CancellationToken)
Microsoft.Build.Unity.<BuildProjectAsync>d__16:MoveNext() (at Library\PackageCache\com.microsoft.msbuildforunity@0.8.2\Editor\ProjectBuilder\MSBuildProjectBuilder.cs:396)
System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1:Start(<BuildProjectAsync>d__16&)
Microsoft.Build.Unity.MSBuildProjectBuilder:BuildProjectAsync(String, BuildEngine, String, IProgress`1, CancellationToken)
Microsoft.Build.Unity.<BuildProjectsAsync>d__10:MoveNext() (at Library\PackageCache\com.microsoft.msbuildforunity@0.8.2\Editor\ProjectBuilder\MSBuildProjectBuilder.cs:289)
System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1:Start(<BuildProjectsAsync>d__10&)
Microsoft.Build.Unity.MSBuildProjectBuilder:BuildProjectsAsync(IReadOnlyCollection`1, String, String, IProgress`1, CancellationToken)
Microsoft.Build.Unity.MSBuildProjectBuilder:BuildProjects(IReadOnlyCollection`1, String, String) (at Library\PackageCache\com.microsoft.msbuildforunity@0.8.2\Editor\ProjectBuilder\MSBuildProjectBuilder.cs:173)
Microsoft.Build.Unity.MSBuildProjectBuilder:BuildAllProjects(String, String) (at Library\PackageCache\com.microsoft.msbuildforunity@0.8.2\Editor\ProjectBuilder\MSBuildProjectBuilder.cs:121)
Microsoft.Build.Unity.MSBuildProjectBuilder:TryBuildAllProjects(String, String) (at Library\PackageCache\com.microsoft.msbuildforunity@0.8.2\Editor\ProjectBuilder\MSBuildProjectBuilderEditor.cs:30)
Microsoft.Build.Unity.ProjectGeneration.MSBuildTools:RefreshGeneratedOutput(Boolean) (at Library\PackageCache\com.microsoft.msbuildforunity@0.8.2\Editor\ProjectGenerator\Scripts\MSBuildTools.cs:177)
Microsoft.Build.Unity.ProjectGeneration.<>c:<.cctor>g__OnUpdate|18_0() (at Library\PackageCache\com.microsoft.msbuildforunity@0.8.2\Editor\ProjectGenerator\Scripts\MSBuildTools.cs:149)
UnityEditor.EditorApplication:Internal_CallUpdateFunctions() (at C:\buildslave\unity\build\Editor\Mono\EditorApplication.cs:200)
 
(Filename: Library/PackageCache/com.microsoft.msbuildforunity@0.8.2/Editor/ProjectBuilder/MSBuildProjectBuilder.cs Line: 403)

Refresh: detecting if any assets need to be imported or removed ...

Refresh Completed time: 0.072013s
	Asset Scan time: 0.068356s
	Asset Hashing: 0.000000s [0 B, 0.000000 mb/s]
	Asset Import (Scripting) time: 0.000000s (count: 0)
	Post Processs Assets (Scripting) time: 0.000000s
	Asset Rehashing: 0.000000s [0 B, 0.000000 mb/s]
	Asset Import (Non Scripting) time: 0.000000s (count: 0)
	Post Process Assets (Non Scripting) time: 0.000000s
	Dependent Assets to Import Queue time: 0.000000s

DirectoryNotFoundException: Could not find a part of the path "H:\New%20Unity%20Project\Library\PackageCache\com.unity.ads@2.0.8\UnityEngine.Advertisements.dll.meta".
  at System.IO.FileStream..ctor (System.String path, System.IO.FileMode mode, System.IO.FileAccess access, System.IO.FileShare share, System.Int32 bufferSize, System.Boolean anonymous, System.IO.FileOptions options) [0x00164] in <e1319b7195c343e79b385cd3aa43f5dc>:0 
  at System.IO.FileStream..ctor (System.String path, System.IO.FileMode mode, System.IO.FileAccess access, System.IO.FileShare share, System.Int32 bufferSize, System.IO.FileOptions options, System.String msgPath, System.Boolean bFromProxy, System.Boolean useLongPath, System.Boolean checkHost) [0x00000] in <e1319b7195c343e79b385cd3aa43f5dc>:0 
  at (wrapper remoting-invoke-with-check) System.IO.FileStream..ctor(string,System.IO.FileMode,System.IO.FileAccess,System.IO.FileShare,int,System.IO.FileOptions,string,bool,bool,bool)
  at System.IO.StreamReader..ctor (System.String path, System.Text.Encoding encoding, System.Boolean detectEncodingFromByteOrderMarks, System.Int32 bufferSize, System.Boolean checkHost) [0x00067] in <e1319b7195c343e79b385cd3aa43f5dc>:0 
  at System.IO.StreamReader..ctor (System.String path, System.Text.Encoding encoding, System.Boolean detectEncodingFromByteOrderMarks, System.Int32 bufferSize) [0x00000] in <e1319b7195c343e79b385cd3aa43f5dc>:0 
  at System.IO.StreamReader..ctor (System.String path, System.Boolean detectEncodingFromByteOrderMarks) [0x0000d] in <e1319b7195c343e79b385cd3aa43f5dc>:0 
  at System.IO.StreamReader..ctor (System.String path) [0x00000] in <e1319b7195c343e79b385cd3aa43f5dc>:0 
  at (wrapper remoting-invoke-with-check) System.IO.StreamReader..ctor(string)
  at Microsoft.Build.Unity.ProjectGeneration.PluginAssemblyInfo.ParseYAMLFile () [0x00007] in H:\New Unity Project\Library\PackageCache\com.microsoft.msbuildforunity@0.8.2\Editor\ProjectGenerator\Scripts\PluginAssemblyInfo.cs:78 
  at Microsoft.Build.Unity.ProjectGeneration.PluginAssemblyInfo..ctor (Microsoft.Build.Unity.ProjectGeneration.UnityProjectInfo unityProjectInfo, System.Guid guid, System.String fullPath, Microsoft.Build.Unity.ProjectGeneration.PluginType type) [0x00032] in H:\New Unity Project\Library\PackageCache\com.microsoft.msbuildforunity@0.8.2\Editor\ProjectGenerator\Scripts\PluginAssemblyInfo.cs:69 
  at Microsoft.Build.Unity.ProjectGeneration.UnityProjectInfo.ScanForPluginDLLs () [0x00171] in H:\New Unity Project\Library\PackageCache\com.microsoft.msbuildforunity@0.8.2\Editor\ProjectGenerator\Scripts\UnityProjectInfo.cs:336 
  at Microsoft.Build.Unity.ProjectGeneration.UnityProjectInfo.RefreshPlugins () [0x00001] in H:\New Unity Project\Library\PackageCache\com.microsoft.msbuildforunity@0.8.2\Editor\ProjectGenerator\Scripts\UnityProjectInfo.cs:88 
  at Microsoft.Build.Unity.ProjectGeneration.UnityProjectInfo..ctor (System.Collections.Generic.HashSet`1[T] supportedBuildTargets) [0x000a2] in H:\New Unity Project\Library\PackageCache\com.microsoft.msbuildforunity@0.8.2\Editor\ProjectGenerator\Scripts\UnityProjectInfo.cs:75 
  at Microsoft.Build.Unity.ProjectGeneration.MSBuildTools.get_UnityProjectInfo () [0x00000] in H:\New Unity Project\Library\PackageCache\com.microsoft.msbuildforunity@0.8.2\Editor\ProjectGenerator\Scripts\MSBuildTools.cs:89 
  at Microsoft.Build.Unity.ProjectGeneration.MSBuildTools.RefreshGeneratedOutput (System.Boolean forceGenerateEverything) [0x0006a] in H:\New Unity Project\Library\PackageCache\com.microsoft.msbuildforunity@0.8.2\Editor\ProjectGenerator\Scripts\MSBuildTools.cs:187 
  at Microsoft.Build.Unity.ProjectGeneration.MSBuildTools+<>c.<.cctor>g__OnUpdate|18_0 () [0x00001] in H:\New Unity Project\Library\PackageCache\com.microsoft.msbuildforunity@0.8.2\Editor\ProjectGenerator\Scripts\MSBuildTools.cs:149 
  at (wrapper delegate-invoke) <Module>.invoke_void()
  at UnityEditor.EditorApplication.Internal_CallUpdateFunctions () [0x00010] in C:\buildslave\unity\build\Editor\Mono\EditorApplication.cs:200 
 
(Filename: <e1319b7195c343e79b385cd3aa43f5dc> Line: 0)

H:\New Unity Project\Assets\MixedRealityToolkit.Providers\WindowsMixedReality\DotNetAdapter> dotnet msbuild DotNetAdapter.csproj -restore  -v:minimal -p:NuGetInteractive=true  -t:Clean -p:Configuration=Release
UnityEngine.DebugLogHandler:Internal_Log(LogType, String, Object)
UnityEngine.DebugLogHandler:LogFormat(LogType, Object, String, Object[])
UnityEngine.Logger:Log(LogType, Object)
UnityEngine.Debug:Log(Object)
Microsoft.Build.Unity.<ExecuteMSBuildAsync>d__17:MoveNext() (at Library\PackageCache\com.microsoft.msbuildforunity@0.8.2\Editor\ProjectBuilder\MSBuildProjectBuilder.cs:403)
System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1:Start(<ExecuteMSBuildAsync>d__17&)
Microsoft.Build.Unity.MSBuildProjectBuilder:ExecuteMSBuildAsync(String, String, String, IProgress`1, CancellationToken)
Microsoft.Build.Unity.<BuildProjectAsync>d__16:MoveNext() (at Library\PackageCache\com.microsoft.msbuildforunity@0.8.2\Editor\ProjectBuilder\MSBuildProjectBuilder.cs:396)
System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1:Start(<BuildProjectAsync>d__16&)
Microsoft.Build.Unity.MSBuildProjectBuilder:BuildProjectAsync(String, BuildEngine, String, IProgress`1, CancellationToken)
Microsoft.Build.Unity.<BuildProjectsAsync>d__10:MoveNext() (at Library\PackageCache\com.microsoft.msbuildforunity@0.8.2\Editor\ProjectBuilder\MSBuildProjectBuilder.cs:289)
System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1:Start(<BuildProjectsAsync>d__10&)
Microsoft.Build.Unity.MSBuildProjectBuilder:BuildProjectsAsync(IReadOnlyCollection`1, String, String, IProgress`1, CancellationToken)
Microsoft.Build.Unity.MSBuildProjectBuilder:BuildProjects(IReadOnlyCollection`1, String, String) (at Library\PackageCache\com.microsoft.msbuildforunity@0.8.2\Editor\ProjectBuilder\MSBuildProjectBuilder.cs:173)
Microsoft.Build.Unity.MSBuildProjectBuilder:BuildAllProjects(String, String) (at Library\PackageCache\com.microsoft.msbuildforunity@0.8.2\Editor\ProjectBuilder\MSBuildProjectBuilder.cs:121)
Microsoft.Build.Unity.MSBuildProjectBuilder:TryBuildAllProjects(String, String) (at Library\PackageCache\com.microsoft.msbuildforunity@0.8.2\Editor\ProjectBuilder\MSBuildProjectBuilderEditor.cs:30)
Microsoft.Build.Unity.ProjectGeneration.MSBuildTools:RefreshGeneratedOutput(Boolean) (at Library\PackageCache\com.microsoft.msbuildforunity@0.8.2\Editor\ProjectGenerator\Scripts\MSBuildTools.cs:177)
Microsoft.Build.Unity.ProjectGeneration.<>c:<.cctor>g__OnUpdate|18_0() (at Library\PackageCache\com.microsoft.msbuildforunity@0.8.2\Editor\ProjectGenerator\Scripts\MSBuildTools.cs:149)
UnityEditor.EditorApplication:Internal_CallUpdateFunctions() (at C:\buildslave\unity\build\Editor\Mono\EditorApplication.cs:200)
```

## Expected behavior

The editor should not get into a  compile loop if MSBuildForUnity is enabled.

## Your Setup (please complete the following information)

- Unity Version 2018.4.12f1
- MRTK Version 2.2 prerelease

**Comment by julenka** [comment]
@davidkline-ms mentioned that this is a bug with MSBuildForUnity, but I wanted to create an issue to track it here so we can reference it in our repo.

**Comment by julenka** [comment]
A possible solution is to mention the following in the release notes:

**If you are using unity remoting you need:**
1) Latest version of remoting player from the store
2) Latest version of Windows Mixed Reality unity package
3) Your project must not have spaces

**Comment by davidkline-ms** [comment]
I can reproduce the errors related to a path with spaces. I cannot cause the infinite loop to reproduce.

**Comment by davidkline-ms** [comment]
This has been resolved with v0.8.3 of MSBuild for Unity. MRTK will serialize this version when enabling msbuild in the configuration dialog, per #6823
